lpc55_sign/src/debug_auth.rs

```diff
@@ -0,0 +1,205 @@
+use lpc55_areas::{CCSOCUDflt, CCSOCUPin, DebugSettings};
+use num_traits::ToPrimitive;
+use rsa::{PublicKeyParts, RsaPrivateKey, RsaPublicKey};
+use sha2::{Digest, Sha256};
+use x509_cert::Certificate;
+
+use crate::{
+    cert::public_key,
+    signed_image::{pad_roots, root_key_hash},
+    Error,
+};
+
+const SOCC: u32 = 0x0000_0001;
+
+#[derive(Debug)]
+pub struct DebugAuthChallenge {
+    version_major: u16,
```

Comment:
The datasheet treats firmware as a single `version: u32`, which might be more clear.

(If you want to keep separate major/minor versions, I think they are currently swapped)

Reply:
I need to add a comment around this.  Yes the datasheet says this is a single, little-endian u32 and then give you 32-bit hex constants for v1.0 and v1.1.  This is _not_ what their tool (https://github.com/nxp-mcuxpresso/spsdk/) generates.  After digging around way too much, I found https://github.com/nxp-mcuxpresso/spsdk/blob/5da31d96a020bd65e5834ea5ac1b68327ea965ef/spsdk/dat/debug_credential.py#L728 which confirms that they actually write this as two, little-endian u16s with the major version at the lower offset.

Reply:
Comments added